crates/ark/src/data_explorer/column_profile.rs

```diff
@@ -0,0 +1,307 @@
+use amalthea::comm::data_explorer_comm::FormatOptions;
+use amalthea::comm::data_explorer_comm::GetColumnProfilesParams;
+use amalthea::comm::data_explorer_comm::ReturnColumnProfilesParams;
+use amalthea::socket::comm::CommSocket;
+use anyhow::anyhow;
+use harp::exec::RFunction;
+use harp::exec::RFunctionExt;
+use harp::tbl_get_column;
+use harp::RObject;
+use harp::TableKind;
+use stdext::unwrap;
+
+use crate::data_explorer::histogram;
+use crate::data_explorer::summary_stats::summary_stats;
+use crate::data_explorer::table::Table;
+use crate::data_explorer::utils::display_type;
+use crate::modules::ARK_ENVS;
+
+pub struct ProcessColumnsProfilesParams {
+    pub table: Table,
+    pub indices: Option<Vec<i32>>,
+    pub kind: TableKind,
+    pub request: GetColumnProfilesParams,
+}
+
+pub async fn handle_columns_profiles_requests(
+    params: ProcessColumnsProfilesParams,
+    comm: CommSocket,
+) -> anyhow::Result<()> {
+    let callback_id = params.request.callback_id;
+    let n_profiles = params.request.profiles.len();
+
+    let profiles = process_columns_profiles_requests(
+        params.table,
+        params.indices,
+        params.kind,
+        params.request.profiles,
+        params.request.format_options,
+    )
+    .await
+    .unwrap_or_else(|e| {
+        // In case something goes wrong whhile computing the profiles, we send
+        // an empty response. Ideally, we would have a way to comunicate an that
+        // an error happened but it's not implemented yet.
+        log::error!("Error while producing profiles: {e}");
+        std::iter::repeat(empty_column_profile_result())
+            .take(n_profiles)
+            .collect()
+    });
+
+    let event = DataExplorerFrontendEvent::ReturnColumnProfiles(ReturnColumnProfilesParams {
+        callback_id,
+        profiles,
+    });
+
+    let json_event = serde_json::to_value(event)?;
+    comm.outgoing_tx.send(CommMsg::Data(json_event))?;
+    Ok(())
+}
+
+async fn process_columns_profiles_requests(
+    table: Table,
+    indices: Option<Vec<i32>>,
+    kind: TableKind,
+    profiles: Vec<ColumnProfileRequest>,
+    format_options: FormatOptions,
+) -> anyhow::Result<Vec<ColumnProfileResult>> {
+    // This is an R thread, so we can actually get the data frame.
+    // If it fails we quickly return an empty result set and end the task.
+    // This might happen if the task was spawned but the data explorer windows
+    // was later closed, before the task actually executed.
+    let data = table.get()?;
+    let mut results: Vec<ColumnProfileResult> = Vec::with_capacity(profiles.len());
+
+    for profile in profiles.into_iter() {
+        log::trace!("Processing column!");
+        results.push(
+            profile_column(
+                data.clone(),
+                indices.clone(),
+                profile,
+                &format_options,
+                kind,
+            )
+            .await,
+        );
+        // Yield to the R event loop
```

Comment:
```suggestion
        // Yield to the idle event loop
```

By R event loop we generally mean the one that runs at interrupt time.

Reply:
Oh you referred to "R console loop" below which is good too.